pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -232,6 +242,156 @@ func getMachineDefaulterOperation(platformStatus *osconfigv1.PlatformStatus) mac
 	}
 }
 
+// NewValidatingWebhookConfiguration creates empty validation webhook configuration
+func NewValidatingWebhookConfiguration() *admissionregistrationv1.ValidatingWebhookConfiguration {
+	return &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: v1.ObjectMeta{
+			Name: DefaultWebhookConfigurationName,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+	}
+}
+
+// WithValidatingHooks returns validating webhooks list to populate configuration
```

Comment:
this name `WithValidatingHooks` seems confusing to me. This is returning a `ValidatingWebhook`, why don't we just name it `ValidatingWebhook()`

Reply:
It is used in `sync.go`, but by domain belongs here, so exporting this is necessary. I initially desired that the machine and machineSet webhooks will have their own methods, and then one could be added for another machine resource (let's say MHC). Making changes will be easier.

Reply:
>and then one could be added for another machine resource (let's say MHC)

please let's introduce abstractions only when they are necessary. Doing it ahead of time adds unnecessary complexity and makes the code more difficult to understand.

Reply:
I disagree. That is a clear separation in the code, not an `unnecessary complexity` by any means. I wrote this and could assure you that mixed definitions of both valid. and mut. hooks is looking obscure and impossible to debug. The code is well written when it does not need comments. In a case of a list with random webhooks, with possible different configuration, it is magic to pull it off and make change without accidentally wrecking things up. I already simplified definitions enough, and can't see a single reason for holding this off. @enxebre please elaborate on what you assume as unnecessary complexity.